packages/trader/src/AppV2/Containers/Trade/trade-types.tsx

```diff
@@ -1,28 +1,230 @@
 import React from 'react';
 import { useTraderStore } from 'Stores/useTraderStores';
-import { Chip, Text } from '@deriv-com/quill-ui';
+import { Chip, Text, ActionSheet } from '@deriv-com/quill-ui';
+import { DraggableList } from 'AppV2/Components/DraggableList';
+import { TradeTypeList } from 'AppV2/Components/TradeTypeList';
 import { getTradeTypesList } from 'AppV2/Utils/trade-types-utils';
+import { Localize, localize } from '@deriv/translations';
 
-type TTemporaryTradeTypesProps = {
+type TTradeTypesProps = {
     onTradeTypeSelect: (e: React.MouseEvent<HTMLButtonElement>) => void;
     trade_types: ReturnType<typeof getTradeTypesList>;
 } & Pick<ReturnType<typeof useTraderStore>, 'contract_type'>;
 
-const TemporaryTradeTypes = ({ contract_type, onTradeTypeSelect, trade_types }: TTemporaryTradeTypesProps) => {
+const TradeTypes = ({ contract_type, onTradeTypeSelect, trade_types }: TTradeTypesProps) => {
+    const [is_open, setIsOpen] = React.useState<boolean>(false);
+    const [is_editing, setIsEditing] = React.useState<boolean>(false);
+
+    const { onMount, onUnmount } = useTraderStore();
+
+    type TItem = {
+        id: string;
+        title: string;
+        icon?: React.ReactNode;
+    };
+
+    type TResultItem = {
+        id: string;
+        title?: string;
+        button_title?: string;
+        onButtonClick?: () => void;
+        items: TItem[];
+    };
+
+    const createArrayFromCategories = (data: any): TItem[] => {
+        const result: TItem[] = [];
+
+        data.forEach((category: { value: string; text: string }) => {
+            result.push({
+                id: category.value,
+                title: category.text,
+            });
+        });
+
+        return result;
+    };
+
+    const saved_other_trade_types = localStorage.getItem('other_trade_types');
+    const saved_pinned_trade_types = localStorage.getItem('pinned_trade_types');
```

Comment:
what do you think about using `JSON.parse` here once instead of multiple times below? :)
```
JSON.parse(localStorage.getItem('pinned_trade_types') ?? '[]')
```